mongo/mongo_test.go

```diff
@@ -385,204 +350,43 @@ func (s *MongoSuite) testEnsureServerNUMACtl(c *gc.C, setNUMAPolicy bool) string
 	return dataDir
 }
 
-func (s *MongoSuite) TestInstallMongod(c *gc.C) {
-	type installs struct {
-		series string
-		cmd    [][]string
-	}
-
-	tests := []installs{
-		{"precise", [][]string{{"--target-release", "precise-updates/cloud-tools", "mongodb-server"}}},
-		{"trusty", [][]string{{"juju-mongodb"}}},
-		{"xenial", [][]string{{"juju-mongodb3.2"}, {"juju-mongo-tools3.2"}}},
-		{"bionic", [][]string{{"mongodb-server-core"}}},
-	}
-
-	testing.PatchExecutableAsEchoArgs(c, s, "add-apt-repository")
-	testing.PatchExecutableAsEchoArgs(c, s, "apt-get")
-	for _, test := range tests {
-		c.Logf("install for series %v", test.series)
-		dataDir := c.MkDir()
-		s.patchSeries(test.series)
-		_, err := mongo.EnsureServer(makeEnsureServerParams(dataDir))
-		c.Assert(err, jc.ErrorIsNil)
-
-		for _, cmd := range test.cmd {
-			match := append(expectedArgs.AptGetBase, cmd...)
-			testing.AssertEchoArgs(c, "apt-get", match...)
-		}
-	}
-}
-
-var fakeInstallScript = `#!/bin/bash
-if [ $# -lt 1 ]
-then
-        echo "Install fail - not enough arguments"
-        exit 1
-fi
-
-# The package name is the last argument
-package=${@: -1}
-echo $package >> %s
-
-if [ $package == "juju-mongodb" ]
-then
-        echo "Installed successfully!"
-        exit 0
-fi
-
-if [ $package == "mongodb-server" ]
-then
-        echo "Installed successfully!"
-        exit 0
-fi
-
-echo "Unable to locate package $package"
-exit 100
-`
-
-func (s *MongoSuite) TestInstallMongodFallsBack(c *gc.C) {
-	if runtime.GOOS == "windows" {
-		c.Skip("Skipping TestInstallMongodFallsBack as mongo is not installed on windows")
-	}
-
+func (s *MongoSuite) TestInstallMongodOnUbuntuViaApt(c *gc.C) {
 	type installs struct {
-		series string
-		cmd    string
+		series  string
+		pkgs    []string
+		expOpts []string
 	}
 
 	tests := []installs{
-		{"precise", "mongodb-server"},
-		{"trusty", "juju-mongodb"},
-		{"xenial", "juju-mongodb3.2"},
-		{"bionic", "mongodb-server-core"},
+		{"precise", []string{"mongodb-server"}, []string{"--target-release", "precise-updates/cloud-tools"}},
```

Comment:
We probably don't need to test for precise anymore, we could arguably even drop trusty for 2.7.